src/core/dbus-execute.c

```diff
@@ -980,12 +981,20 @@ static int property_get_exec_dir_symlink(
         if (r < 0)
                 return r;
 
-        FOREACH_ARRAY(i, d->items, d->n_items)
+        FOREACH_ARRAY(i, d->items, d->n_items) {
+                /* The old exec directory properties cannot represent flags, so list them here with no
+                 * destination */
+                if ((i->flags & _EXEC_DIRECTORY_FLAGS_PUBLIC) != 0 && !strv_isempty(i->symlinks)) {
```

Comment:
the strv_isempty() check seems to be reversed?

also, here too, please drop the flag check. Let's just consider the symlink property a superset of the non-symlink property from now on.

Reply:
> the strv_isempty() check seems to be reversed?

it's not, the intention is: if there are symlinks, then it will be returned in the loop below. If it's a case of directory + r/o but no symlinks, it will not be added to the list below since the array is empty, so it has to be processed here